plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/SocketAccess.java

```diff
@@ -44,33 +44,51 @@ private SocketAccess() {}
         try {
             return AccessController.doPrivileged(operation);
         } catch (PrivilegedActionException e) {
-            throw (IOException) e.getCause();
+            Throwable cause = e.getCause();
+            if (cause instanceof IOException) {
+                throw (IOException) e.getCause();
+            } else {
+                throw new IOException(cause);
```

Comment:
It feels wrong to wrap this as an IOException. Can we just use `Throwables.rethrow(cause)` here?

Reply:
I'd rather not tbh. We could wrap in a different exception but `Throwables.rethrow` itself just casts to `RunTimeException` doesn't it? In the end if the SDK ever changes or we missed some checked exception in our code, we'll get the same bug we started from here again won't we?

Reply:
> but Throwables.rethrow itself just casts to RunTimeException doesn't it

no it does not

Reply:
Huh, Java works in mysterious ways. Sorry for the noise, my bad seems it works just fine :)
=> used it all the way now.